### DISCUSSION TITLE
Server running on PHP 8

### DISCUSSION THREAD

**Discussion by usernotnull** [enlightn/enlightn#31] [Q&A] [answered]
Does this lib integrate with a server running on PHP 8?

I got the following:
 Problem 1
    - enlightn/security-checker[v1.1, ..., v1.4] require ext-zip * -> it is missing from your system. Install or enable PHP's zip extension.
    - enlightn/enlightn v1.8.0 requires enlightn/security-checker ^1.1 -> satisfiable by enlightn/security-checker[v1.1, v1.2, v1.3, v1.4].
    - Root composer.json requires enlightn/enlightn ^1.8 -> satisfiable by enlightn/enlightn[v1.8.0].

I contacted hosting support and they mentioned that this extension is not available in PHP 8.

Thanks

**Comment by paras-malhotra** [comment, marked as answer]
@RJFares, yes PHP 8 is supported by Enlightn. In fact our tests run on PHP 7.3, 7.4 and 8.0.

https://github.com/enlightn/enlightn/blob/3755969ae7c0149c4ec2143a39fe8b4a96d909cf/.github/workflows/tests.yaml#L12-L14

You need to install the zip PHP 8 extension. I can find the debian package [here](https://packages.debian.org/sid/main/php8.0-zip) so I do think it's available for PHP 8.

**Reply by usernotnull** [reply]
I have contacted support and they mentioned that I'm on shared hosting and at the moment they cannot install custom extensions on their RHEL Centos 7 so I'm stuck with that for the moment until they decide to install it (or downgrade the PHP version to one that has the PHP zip extension which I won't be doing).

Thanks for the reply.

**Reply by paras-malhotra** [reply]
@RJFares, the zip extension is no longer required. Just wanted to update you on this.

**Reply by usernotnull** [reply]
@paras-malhotra thanks for the update. In fact, I received news just last week that our servers now installed zip extension :)
Will be implementing it soon.
